### PR TITLE
EZP-32277: Support for setting up Trusted Proxies via ENVs or symfony semantic configuration

### PR DESCRIPTION
**Pull request by DominikaK**
| Question      | Answer
| ------------- | ---
| JIRA Ticket   | https://issues.ibexa.co/browse/EZP-32277
| Versions      | 3.3, master

Deprecation notice in v3.3 release notes: https://github.com/ezsystems/developer-documentation/pull/1263/commits/f65f9b9491116648303aaf70750719ea984fbec6

### CODE REVIEW

**Review by ViniTou — APPROVED** [review]
It's ok for current state, I wonder if we should also somewhere mention that _just_ setting TRUSTED_PROXIES env will stop working.

Also, line 152 with `In a pure Symfony installation you would normally adapt front controller (`web/app.php`)`

is no longer true, as there is no front controller in `web/app.php` and you set those things up by semnatic configuration.

File: docs/guide/http_cache.md [hunk 1]
```diff
@@ -158,6 +158,13 @@ however in [[= product_name =]] can cover most use cases by setting supported en
...
 - `TRUSTED_PROXIES`: String with trusted IP, multiple proxies can be configured with a comma, i.e. `TRUSTED_PROXIES="192.0.0.1,10.0.0.0/8"`
 
+Add the trusted proxies to your configuration:
+
+``` yaml
+framework:
+    trusted_proxies: '%env(TRUSTED_PROXIES)%'
+```
+
 !!! caution "Careful when trusting dynamic IP using TRUST_REMOTE value or similar"
```
Comment (ViniTou) [inline docs/guide/http_cache.md:168]: Also, `TRUST_REMOTE` is no longer with us, long live for `REMOTE_ADDR`.
https://github.com/ezsystems/ezplatform/pull/537

**Comment by DominikaK** [comment]
>we should also somewhere mention that just setting TRUSTED_PROXIES env will stop working.

This is covered in the releases notes.

File: docs/guide/http_cache.md [hunk 2]
```diff
@@ -149,19 +149,25 @@ Configuring [[= product_name =]] for Varnish or Fastly involves a few steps, sta
 
 #### Configuring Symfony front controller
 
-In a pure Symfony installation you would normally adapt front controller (`web/app.php`)
-in order to configure Symfony to [work behind a load balancer or a reverse proxy](https://symfony.com/doc/5.1/deployment/proxies.html),
-however in [[= product_name =]] can cover most use cases by setting supported environment variables using:
+In order to configure Symfony to [work behind a load balancer or a reverse proxy](https://symfony.com/doc/5.1/deployment/proxies.html),
+set supported environment variables using:
```
Comment (dabrt) [inline docs/guide/http_cache.md:153]:
```suggestion
set the supported environment variables by using the following parameters:
```

File: docs/guide/http_cache.md [hunk 3]
```diff
@@ -149,19 +149,25 @@ Configuring [[= product_name =]] for Varnish or Fastly involves a few steps, sta
...
+``` yaml
+framework:
+    trusted_proxies: '%env(TRUSTED_PROXIES)%'
+```
+
+!!! caution "Careful when trusting dynamic IP using REMOTE_ADDR value or similar"
 
     On Platform.sh, Varnish does not have a static IP, like with [AWS LB.](https://symfony.com/doc/5.1/deployment/proxies.html#but-what-if-the-ip-of-my-reverse-proxy-changes-constantly)
-    For this `TRUSTED_PROXIES` env variable supports being set to value "TRUST_REMOTE", which is equal to:
+    For this `TRUSTED_PROXIES` env variable supports being set to value "REMOTE_ADDR", which is equal to:
```
Comment (dabrt) [inline docs/guide/http_cache.md:170]:
```suggestion
    For this reason, the `TRUSTED_PROXIES` env variable supports being set to value "REMOTE_ADDR", which is equal to:
```
Or "therefore"?

**Review by dabrt — APPROVED** [review]

**Pull request merged**